### ISSUE TITLE
`fatal error: 'gnu/stubs-32.h' file not found` while building

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#368]
If you are seeing the following or a similar error while building Cilium
or while building programs for endpoints:
```
    In file included from /usr/include/features.h:392:
    /usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
    # include <gnu/stubs-32.h>
              ^
    1 error generated.
```
Then your system does not have the 32-bit versions of Glibc installed. Install
them via your distribution's package, e.g. for Fedora:
```
    dnf install glibc-devel.i686
```

**Comment by aanm** [comment]
In ubuntu `apt-get install libc6-dev-i386`

**Comment by mrostecki** [comment]
I think we should re-open it and fix properly, in relation to #8529

The content of `/usr/include/gnu/stubs.h` is usually:

```
#if !defined __x86_64__
# include <gnu/stubs-32.h>
#endif
#if defined __x86_64__ && defined __LP64__
# include <gnu/stubs-64.h>
#endif
#if defined __x86_64__ && defined __ILP32__
# include <gnu/stubs-x32.h>
#endif
```

If we are compiling Cilium on 64-bit architecture, we should have __x86_64__ and __LP64__ macros defined, but for some reason we don't have them - and finding that reason should be the goal of this issue. There is (most probably) no reason to use 32-bit headers of glibc and install multilib glibc.

**Comment by nanfengnan1** [comment]
ok, kids! i have a qz, if i compile program using cross compile toolchains, occur this problem, how can i do?
